protocols/raft/src/main/java/io/atomix/protocols/raft/RaftServer.java

```diff
@@ -598,6 +596,17 @@ public Builder withName(String name) {
       return this;
     }
 
+    /**
+     * Sets the cluster service.
+     *
+     * @param clusterService the cluster service
+     * @return the server builder
+     */
+    public Builder withClusterService(ClusterService clusterService) {
+      this.clusterService = checkNotNull(clusterService, "clusterService cannot be null");
```

Comment:
Verify non null in the builder when building?

Reply:
It's done in the `RaftContext` constructor